pkg/aaq-server/handler/handler.go

```diff
@@ -259,3 +220,63 @@ func getResourcesNames(resourceList v1.ResourceList) []v1.ResourceName {
 	}
 	return keys
 }
+
+func generatePatchesForPodWithNodeName(pod v1.Pod) []patch.PatchOperation {
+	if pod.Spec.NodeName == "" {
+		return nil
+	}
+
+	affinity := pod.Spec.Affinity
+	affinityPatchOp := patch.PatchReplaceOp
+
+	if affinity == nil {
+		affinity = &v1.Affinity{}
+		affinityPatchOp = patch.PatchAddOp
+	}
+	if affinity.NodeAffinity == nil {
+		affinity.NodeAffinity = &v1.NodeAffinity{}
+	}
+	if affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+		affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &v1.NodeSelector{}
+	}
+	affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(
+		affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms,
+		v1.NodeSelectorTerm{
+			MatchFields: []v1.NodeSelectorRequirement{
+				{
+					Key:      "metadata.name",
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{pod.Spec.NodeName},
+				},
+			},
+		},
+	)
+
+	tolerations := pod.Spec.Tolerations
+	if tolerations == nil {
+		tolerations = []v1.Toleration{}
+	}
+	tolerations = append(tolerations, v1.Toleration{
+		Key:      "", // i.e. any key
+		Operator: v1.TolerationOpExists,
+		Effect:   v1.TaintEffectNoSchedule,
+	})
+
+	return []patch.PatchOperation{
+		{
+			Op:    affinityPatchOp,
```

Comment:
we can use patch.PatchReplaceOp directly :)

Reply:
Done!